### ISSUE TITLE
Re-Investigate Swapping Webpack for ESBuild Preprocessor

### ISSUE THREAD

**Issue by pjhill** [department-of-veterans-affairs/va.gov-team#26317]
# Description
After the initial investigation in this [ticket](https://app.zenhub.com/workspaces/vsp-5cedc9cce6e3335dc5a49fc4/issues/department-of-veterans-affairs/va.gov-team/23565), we determined that we need to investigate this again once more development has taken place with ESBuild. Let's attempt the swap again in Q4 when ESBuild has potentially reached version 1.0.

## Tasks
- [x] Attempt to swap the latest version of ESBuild in for Webpack
- [x] Troubleshoot any issues that arise
- [x] Create a test branch and feed PRs into it to spot check that the ESBuild version of the Cypress run produces identical results to Webpack
- [x] Collect results of testing and report

## Acceptance Criteria
- [x] ESBuild produces identical test results to a version of the Cypress E2E tests using Webpack
- [x] The swap for ESBuild results in faster execution times in our GitHub Actions CI/CD pipeline

**Comment by ddzz** [comment]
The default ESBuild configuration does not work with the Cypress form tester due to issues with importing Node.js functions. There is no simple solution to this as far as I can tell. This ticket is being moved back to the backlog so that we can. reinvestigate this when ESBuild is more mature.

**Comment by ddzz** [comment]
Yesterday I found a discussion on GitHub that had a fix for the import issue I ran into in the form tester. This reduced the number of failing tests from around 60 to 18. I fixed five of those 18 after addressing issues with missing global variables and the app registry not being configured properly.

The remaining 13 failures appear to be caused by unnecessary imports in some Cypress tests. Today I will look into how to exclude these unneeded files and then check how many failures remain.

**Comment by ddzz** [comment]
12 of the 13 remaining failures have been fixed. WIP branch: https://github.com/department-of-veterans-affairs/vets-website/compare/cypress-esbuild

**Comment by ddzz** [comment]
Update: the last failure has been fixed. The next step is to verify that the recent fixes did not cause issues in any other tests. After that I will work on getting this working in CI and doing additional testing.

**Comment by ddzz** [comment]
I was able to get tests running and passing in CI. There is a warning appearing for videos generated for skipped tests; I'm looking into fixing this. After I fix this warning I will test various scenarios in CI.

**Comment by ddzz** [comment]
The warning for videos was caused by tests videos being too short to be processed due to the performance improvements from using ESBuild. [This workaround](https://github.com/cypress-io/cypress/issues/20012#issuecomment-1028034878) fixed it.

Today I tested multiple aspects of Cypress testing:
 * aXe checks
 * generating videos
 * generating screenshots
 * test selection

All scenarios functioneed as  expected.

ESBuild reduces CI times by approximately three minutes when the full test suite runs.
